boards/.cargo/config.toml

```diff
@@ -0,0 +1,27 @@
+# Licensed under the Apache License, Version 2.0 or the MIT License.
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+# Copyright Tock Contributors 2024.
+
+[build]
+rustflags = [
+  # Use the LLVM lld executable with the `-flavor gnu` flag.
+  "-C", "linker-flavor=ld.lld",
+  # Use static relocation model. See https://github.com/tock/tock/pull/2853
+  "-C", "relocation-model=static",
+  # lld by default uses a default page size to align program sections. Tock
+  # expects that program sections are set back-to-back. `-nmagic` instructs the
+  # linker to not page-align sections.
+  "-C", "link-arg=-nmagic",
+  # Identical Code Folding (ICF) set to all. This tells the linker to be more
+  # aggressive about removing duplicate code. The default is `safe`, and the
+  # downside to `all` is that different functions in the code can end up with
+  # the same address in the binary. However, it can save a fair bit of code
+  # size.
+  "-C", "link-arg=-icf=all",
```

Comment:
Let's just move these to build.rs too. Or at least be clear about doing one or the other.

Reply:
Can't, as far as I know. `build.rs`  only supports adding a subset of `RUSTFLAGS`. I agree that's unfortunate. I moved as much as I could figure out how to move.

Reply:
Does https://doc.rust-lang.org/cargo/reference/build-scripts.html#rustc-link-arg only work with some sub flags? It seems like we should be able to do the `-C link-arg=` ones.

Reply:
Will try!